### ISSUE TITLE
Refactor `OrbitalRotation` class for `ooVQE`

### ISSUE THREAD

**Issue by ElePT** [Qiskit/qiskit-nature#575]
### What is the expected enhancement?

As stated in issue #124, the `ooVQE` algorithm needs to undergo a major refactoring to fit into the current Qiskit Nature framework. This algorithm is an extension of VQE where we optimize the rotation matrix of one body integrals and the ansatz parameters simultaneously. The rotation matrix is implemented by the `OrbitalRotation` class, which also needs to be refactored. 

This refactoring includes:

- [ ] removal of `FermionicTransformation`, and use of the current `QubitConverter`/ `problems` framework instead
- [ ] review of freeze-core support following the issue pointed out in #124 (this issue/code is a bit outdated, so maybe it will work fine)
- [ ] minor changes regarding error types, etc. to match the current Qiskit Nature framework

Once the `OrbitalRotation` class is finished, the next step is to refactor the rest of the ooVQE algorithm, which will be tracked in a separate issue.
